File: 3713_Longest_Balanced_Substring_I/solution.c
int longestBalanced(char* s) {
    int n = strlen(s), result = 0;

    for (int i = 0; i < n; i++) {
        int freq[26] = {0};
        for (int j = i; j < n; j++) {
            freq[s[j] - 'a']++;
            int first = 0, ok = 1;
            for (int k = 0; k < 26; k++) {
                if (freq[k] > 0) {
                    if (first == 0) first = freq[k];
                    else if (freq[k] != first) { ok = 0; break; }
                }
            }
            if (ok) { int len = j - i + 1; result = (len > result) ? len : result; }
        }
    }
    return result;
}

File: 3713_Longest_Balanced_Substring_I/solution.py
class Solution:
    def longestBalanced(self, s: str) -> int:
        n, result = len(s), 0
        for i in range(n):
            count = {}
            for j in range(i, n):
                count[s[j]] = count.get(s[j], 0) + 1
                freq = list(count.values())
                if len(set(freq)) == 1:
                    result = max(result, j - i + 1)

        return result